cime/config/e3sm/machines/config_compilers.xml

```diff
@@ -78,6 +78,12 @@ for mct, etc.
      compilers -->
 <compiler>
   <SUPPORTS_CXX>FALSE</SUPPORTS_CXX>
+  <CPPDEFS>
+    <!-- MPAS Components rely on this flag to use the newer PIO/Scorpio interfaces. These newer interfaces are available on both Scorpio and Scorpio classic -->
+    <append MODEL="ice"> -DUSE_PIO2 </append>
+    <append MODEL="ocn"> -DUSE_PIO2 </append>
+    <append MODEL="glc"> -DUSE_PIO2 </append>
+  </CPPDEFS>
```

Comment:
In this file MODEL is used to refer to components by name, like "cism" or "cam". But this must work to pass all the tests? I'm more curious than anything....

Reply:
Yes, this change sets the flags for only ice, ocn and glc components.

Reply:
I was just curious because if I grep "MODEL" in that file, all the results are model names like "cam" or "clm" instead of component types like atm or lnd:
```[wolf966@compy-e machines]$ grep MODEL config_compilers.xml 
    <append MODEL="moby"> -DDIR=NOOP </append>
    <append MODEL="csm_share"> -std=c99 </append>
    <append MODEL="cism"> -D CISM_GNU=ON </append>
    <append MODEL="cism"> -D CISM_GNU=ON </append>
    <append MODEL="datm"> -Mnovect </append>
    <append MODEL="dlnd"> -Mnovect </append>
    <append MODEL="drof"> -Mnovect </append>
    <append MODEL="dwav"> -Mnovect </append>
    <append MODEL="dice"> -Mnovect </append>
    <append MODEL="docn"> -Mnovect </append>
    <append MODEL="cam"> </append>
    <append MODEL="datm"> -Mnovect </append>
    <append MODEL="dlnd"> -Mnovect </append>
    <append MODEL="drof"> -Mnovect </append>
    <append MODEL="dwav"> -Mnovect </append>
    <append MODEL="dice"> -Mnovect </append>
    <append MODEL="docn"> -Mnovect </append>
```

Reply:
@jgfouca can you confirm that the above usage of MODEL attribute is supported by CIME (MODEL=ice/ocn/glc)?

(It works as expected for me but @jonbob wants to confirm that the usage is correct)

Reply:
@jayeshkrishna , @jonbob , looking at build.py:

```
    with open(file_build, "w") as fd:
        stat = run_cmd("MODEL={} SMP={} {} {} {} {} "
                       .format(compclass, stringify_bool(smp), cmd, caseroot, libroot, bldroot),
                       from_dir=bldroot,  arg_stdout=fd,
                       arg_stderr=subprocess.STDOUT)[0]
```

It looks like MODEL is set to the compclass, so your usage is correct. I'm concerned that all the other uses may be wrong.

Reply:
Thanks @jgfouca

Reply:
I'm doing some tests now to confirm

Reply:
Thanks for following up on this, @jayeshkrishna and @jgfouca . @rljacob , do we need to fix this file so the MODEL usage is consistent?

Reply:
Looking through Tools/Makefile, it appears as though MODEL actually is supposed to be a component name, not component class; this may be a case of a mislabelled variable in build.py. I vaguely remember sorting this stuff out in the CMake work.

Reply:
Unfortunately, it appears there is no guarantee on the contents of MODEL. The default is the comp class, but many of the buildlib scripts for individual components appear to override it with their name. This is one of the things I cleaned-up in the Cmake work.

Reply:
@jgfouca - I think it might be more appropriate for MODEL to be the specific model itself, especially in this context. For example, we might need to add compiler options for mpas-o that would not be correct for docn -- and yet they both would be referred to as component "OCN"

Reply:
@jonbob , I agree. I think we want to support selectors for both model name and type.

Reply:
Looking at how MODEL is used in config_compilers for both cesm and e3sm, its using component names.  It even uses "gptl", "driver" and "csm_share".  So we should continue that here.  @jayeshkrishna please make that change.

Reply:
Has this been resolved?

Reply:
No, I haven't fixed it in this PR (its already merged to next). I can create a new PR with the changes (In this case however the flags work for all relevant model types - e.g. works for docn too).